p5/core/PShape.py

```diff
@@ -0,0 +1,157 @@
+#py#
+# Part of p5: A Python package based on Processing
+# Copyright (C) 2017 Abhik Pal
+# Copyright (C) 2018 Aliabbas Merchant
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+from . import primitives
+from ..pmath import Point, Vector
+import warnings
+
+MIN_POINT_ACCURACY = 20
+MAX_POINT_ACCURACY = 200
+POINT_ACCURACY_FACTOR = 10
+
+OPEN, CLOSE = 1, 2
```

Comment:
Is there a nicer way of doing this other than assigning numbers to constants?

Reply:
We may use strings instead.  
But numbers are faster to work with.  
Also, the variables have been named quite appropriately.  
Plus, these are the same constant names as used in the Java code, so there is a less of a learning curve/ less work while switching from Java to p5.

Reply:
I was wondering for other solutions that do not necessarily use these variables. For instance, instead of defining ```OPEN``` and ```CLOSE``` we could just use keyword arguments: ```def end_shape(self, open=True)```

> We may use strings instead.  But numbers are faster to work with.

Correct. But this doesn't seem to influence things that much. I looked around a bit and at least in the popular scientific computing libraries this seems to be quite common. In scipy for instance, one would often do ```scipy.optimize.minimize(f, x0, method='CG')```. I really don't think something like ```self._kind == 'group'``` would be an issue.

Reply:
Seems like if n1 attempts to convert a Java or Python Mode sketch into p5py is gonna know hell!  :P

Reply:
@GoToLoop, ¯\\\_(ツ)\_/¯